src/main/java/org/mitre/synthea/world/agents/behaviors/planfinder/PlanFinderPriority.java

```diff
@@ -16,22 +15,16 @@ public class PlanFinderPriority implements IPlanFinder {
   public InsurancePlan find(Set<InsurancePlan> plans, Person person,
       EncounterType service, long time) {
 
-    // Government plan selection does not consider affordability.
-    // Private plans require that a person meets basic affordabilty/occupation requirements.
-    List<InsurancePlan> eligiblePlans = (plans.stream().filter(plan ->
-        (plan.isGovernmentPlan() && plan.accepts(person, time))
+    List<InsurancePlan> eligiblePlans = (plans.stream().filter(plan -> (plan.isGovernmentPlan()
         || (IPlanFinder.meetsAffordabilityRequirements(plan, person, service, time)))
-        .collect(Collectors.toList()));
+        && plan.accepts(person, time)).collect(Collectors.toList()));
 
-    if (!eligiblePlans.isEmpty()) {
-      // If there are affordable/eligible plans, choose the ones with the highest priority.
-      Set<Integer> eligiblePriorities = eligiblePlans.stream().map(plan ->
-          plan.getPayer().getPriority()).collect(Collectors.toSet());
-      int maxEligiblePriority = Collections.min(eligiblePriorities);
-      eligiblePlans = eligiblePlans.stream()
-          .filter(plan -> plan.getPayer().getPriority() == maxEligiblePriority)
-          .sorted(Comparator.comparing(InsurancePlan::toString))
-          .collect(Collectors.toList());
+    if (eligiblePlans.size() > 1) {
+      // If there are more than 1 affordable/eligible plans, filter to the highest priority ones.
+      int highestEligibltPriority = eligiblePlans.stream()
```

Comment:
It looks like higher priority plans have a lower value returned by `getPriority`, e.g. medicare has a lower priority value than others but is a higher priority pla?. If I got this right then the name of this variable might be a bit confusing and it would be good to document that low priority values represent high priorities in `InsurancePlan.getPriority` javadoc. Alternative you could reverse it so a larger int value reflects a higher priority. If you do that then you'd need to switch the `min` to `max` here.

Reply:
It seems to comply with the logic underneath the common phrase "My number 1 priority is..."